src/openzaak/components/zaken/api/validators.py

```diff
@@ -639,3 +640,64 @@ def __call__(self, attrs, serializer):
             raise serializers.ValidationError(
                 {"overige_relatie": self.message}, code=self.code
             )
+
+
+class BehandelaarTijdvakGeldigheidValidator:
+    code = "behandelaar-tijdvak-geldigheid-overlap"
+    message = _(
+        "Only one Rol with roltype.omschrijvingGeneriek `behandelaar` can exist "
+        "during a validity period (tijdvakGeldigheid) for betrokkeneType {betrokkene_type}"
+    )
+    requires_context = True
+
+    def __init__(self, betrokkene_type):
+        self.betrokkene_type = betrokkene_type
+
+    def __call__(self, attrs, serializer):
+        betrokkene_type = get_from_serializer_data_or_instance(
+            "betrokkene_type", attrs, serializer
+        )
+        roltype = get_from_serializer_data_or_instance("roltype", attrs, serializer)
+        zaak = get_from_serializer_data_or_instance("zaak", attrs, serializer)
+        tijdvak_geldigheid = get_from_serializer_data_or_instance(
+            "tijdvak_geldigheid", attrs, serializer
+        )
+
+        # Validation does not apply if no tijdvakGeldigheid is supplied, to make it
+        # a backwards compatible experimental feature
+        if (
+            not tijdvak_geldigheid
+            or betrokkene_type != self.betrokkene_type
+            or roltype.omschrijving_generiek != RolOmschrijving.behandelaar
```

Comment:
I think it is like extension of `RolOccurenceValidator` but with dates?

Reply:
I forgot to remove this validator, it was mentioned in refinement that we won't do validation on this, just record allow the validity period to be recorded https://github.com/open-zaak/open-zaak/issues/1809#issuecomment-2647716834